src/core/rate_limiter.rs

```diff
@@ -31,7 +31,7 @@ impl RateLimiter {
         }
     }
 
-    pub fn is_allowed(&mut self, timestamp_ns: f64) -> bool {
+    pub fn _is_allowed(&mut self, timestamp_ns: f64) -> bool {
```

Comment:
```suggestion
    pub fn is_allowed(&mut self, timestamp_ns: f64) -> bool {
```

this one doesn't need to be "private", this API doesn't have to be 1:1 with the Python API since it isn't exposed.